File: src/lib/rubeno/lib/base_feed.rb
module Rubeno
  # BaseFeed extends BaseAction to support TDT (Table Driven Testing) pattern.
  class BaseFeed < BaseAction
    attr_accessor :row_index, :row_data
    
    def initialize(name, feed_cb)
      super(name, feed_cb)
      @row_index = -1
      @row_data = nil
    end
    
    # Set the current row data before processing
    def set_row_data(index, data)
      @row_index = index
      @row_data = data
    end
    
    # Alias perform_action to feed for TDT pattern
    def feed(store, feed_cb, test_resource)
      perform_action(store, feed_cb, test_resource)
    end
    
    # Alias test to process_row for TDT pattern
    def process_row(store, test_resource_configuration, row_index, row_data)
      set_row_data(row_index, row_data)
      test(store, test_resource_configuration)
    end
  end
end

File: src/lib/tiposkripto/src/BaseFeed.ts
import { BaseAction } from "./BaseAction.js";
import { TestTypeParams_any } from "./CoreTypes.js";

/**
 * BaseFeed extends BaseAction to support TDT (Table Driven Testing) pattern.
 * It processes each row from the table.
 */
export class BaseFeed<I extends TestTypeParams_any> extends BaseAction<I> {
  // Row index being processed
  rowIndex: number = -1;
  rowData: any = null;

  constructor(name: string, feedCB: (xyz: I["iselection"]) => I["then"]) {
    super(name, feedCB);
  }

  // Set the current row data before processing
  setRowData(index: number, data: any) {
    this.rowIndex = index;
    this.rowData = data;
  }

  // Alias performAction to feed for TDT pattern
  async feed(
    store: I["istore"],
    feedCB: (x: I["iselection"]) => I["then"],
    testResource
  ) {
    // Pass row data to the callback if needed
    // We can modify the store to include row data
    return super.performAction(store, feedCB, testResource);
  }

  // Alias test to processRow for TDT pattern
  async processRow(
    store: I["istore"],
    testResourceConfiguration,
    rowIndex: number,
    rowData: any
  ) {
    this.setRowData(rowIndex, rowData);
    return super.test(store, testResourceConfiguration);
  }
}

export type IFeeds<I extends TestTypeParams_any> = Record<string, BaseFeed<I>>;
